src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php

```diff
@@ -1227,6 +1252,71 @@ public function findServiceGroupsByHostAndService(int $hostId, int $serviceId):
         return $serviceGroups;
     }
 
+    /**
+     * Find downtimes for host or service
+     * @param int $hostId
+     * @param int $serviceId
+     * @return array
+     */
+    public function findDowntimes(int $hostId, int $serviceId)
+    {
+        $downtimes = [];
+
+        if (empty($hostId)) {
+            return $downtimes;
+        }
+
+        $sql = 'SELECT * FROM `:dbstg`.`downtimes` WHERE host_id = :hostId AND service_id = :serviceId ' .
+                'AND deletion_time IS NULL';
+        $request = $this->translateDbName($sql);
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':hostId', $hostId, \PDO::PARAM_INT);
+        $statement->bindValue(':serviceId', $serviceId, \PDO::PARAM_INT);
+        $statement->execute();
+
+        while (false !== ($result = $statement->fetch(\PDO::FETCH_ASSOC))) {
+            $downtimes[] = EntityCreator::createEntityByArray(
+                Downtime::class,
+                $result
+            );
+        }
+
+        return $downtimes;
+    }
+
+    /**
+     * Find acknowledgements for host or service
+     * @param int $hostId
+     * @param int $serviceId
+     * @return array
+     */
+    public function findAcknowledgements(int $hostId, int $serviceId)
+    {
+        $acks = [];
+
+        if (empty($hostId)) {
+            return $acks;
+        }
+
+        $sql = 'SELECT * FROM `:dbstg`.`acknowledgements` WHERE host_id = :hostId AND service_id = :serviceId ' .
+                'AND deletion_time IS NULL ORDER BY entry_time DESC';
```

Comment:
the same as for https://github.com/centreon/centreon/pull/8398/files#diff-59a4179864ef41b5691108a8d0db941cR1270